libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/transform.hpp

```diff
@@ -303,40 +358,357 @@ namespace hpx { namespace parallel { inline namespace v1 {
     ///       the end of the shorter of the two given input sequences
     ///
     /// \returns  The \a transform algorithm returns a
-    /// \a hpx::future<tagged_tuple<tag::in1(InIter1), tag::in2(InIter2), tag::out(OutIter)> >
+    /// \a hpx::future<ranges::binary_transform_result<
+    ///           typename hpx::traits::range_iterator<Rng1>::type,
+    ///           typename hpx::traits::range_iterator<Rng2>::type,
+    ///           FwdIter> >
     ///           if the execution policy is of type \a parallel_task_policy
     ///           and returns
-    /// \a tagged_tuple<tag::in1(InIter1), tag::in2(InIter2), tag::out(OutIter)>
+    /// \a ranges::binary_transform_result<
+    ///           typename hpx::traits::range_iterator<Rng1>::type,
+    ///           typename hpx::traits::range_iterator<Rng2>::type,
+    ///           FwdIter>
     ///           otherwise.
     ///           The \a transform algorithm returns a tuple holding an iterator
-    ///           referring to the first element         r the first input sequence,
+    ///           referring to the first element after the first input sequence,
     ///           an iterator referring to the first element after the second
     ///           input sequence, and the output iterator referring to the
     ///           element in the destination range, one past the last element
     ///           copied.
     ///
+    template <typename ExPolicy, typename Rng1, typename Rng2, typename FwdIter,
+        typename F, typename Proj1 = hpx::parallel::util::projection_identity,
+        typename Proj2 = hpx::parallel::util::projection_identity>
+    typename parallel::util::detail::algorithm_result<ExPolicy,
+        ranges::binary_transform_result<
+            typename hpx::traits::range_iterator<Rng1>::type,
+            typename hpx::traits::range_iterator<Rng2>::type, FwdIter>>::type
+    tag_invoke(hpx::ranges::transform_t, ExPolicy&& policy, Rng1&& rng1,
+        Rng2&& rng2, FwdIter dest, F&& f, Proj1&& proj1 = Proj1(),
+        Proj2&& proj2 = Proj2())
+
+}    // namespace hpx
+#else    // DOXYGEN
+
+#include <hpx/config.hpp>
+#include <hpx/concepts/concepts.hpp>
+#include <hpx/iterator_support/range.hpp>
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/iterator_support/traits/is_range.hpp>
+#include <hpx/parallel/util/result_types.hpp>
+#include <hpx/parallel/util/tagged_pair.hpp>
+#include <hpx/parallel/util/tagged_tuple.hpp>
```

Comment:
Are the `tagged_xxx.hpp` headers still needed?

Reply:
No they are not. Please don't merge there is still some minor housekeeping left done. It will be ready within the day. Thanks :)